strawberry/permission.py

```diff
@@ -169,7 +170,10 @@ async def resolve_async(
 
             if not has_permission:
                 return self._on_unauthorized(permission)
-        return await next_(source, info, **kwargs)
+        next = next_(source, info, **kwargs)
+        if inspect.isasyncgen(next):
+            return next
+        return await next
```

Comment:
This will do for now, but my favored solution is to have a separate `resolve_asyncgen` next to sync and async resolvers for each extension. Then we can have a compatibility field extension that wraps the asnyc gen in an awaitable to return to classic async extensions.